curvefs/src/client/s3/disk_cache_manager_impl.cpp

```diff
@@ -56,13 +56,37 @@ int DiskCacheManagerImpl::Init(const S3ClientAdaptorOption option) {
 }
 
 void DiskCacheManagerImpl::Enqueue(
-  std::shared_ptr<PutObjectAsyncContext> context) {
+  std::shared_ptr<PutObjectAsyncContext> context, bool isReadWrite) {
+    if (isReadWrite) {
+        auto task = [this, context]() {
+            this->WriteReadDirectClosure(context);
+        };
+        taskPool_.Enqueue(task);
+        return;
+    }
     auto task = [this, context]() {
         this->WriteClosure(context);
     };
     taskPool_.Enqueue(task);
 }
 
+
+
+int DiskCacheManagerImpl::WriteReadDirectClosure(
+    std::shared_ptr<PutObjectAsyncContext> context) {
+        VLOG(9) << "WriteReadClosure start, name: " << context->key;
+        int ret = WriteReadDirect(context->key,
```

Comment:
The write may have to be put after the s3 upload is successful, otherwise if the write is successful but the client exits before the upload is successful, the data will be inconsistent